### PR TITLE
[NTUSER] menu.c position fix to not hide 'Close'

### PR DESCRIPTION
**Pull request by Doug-Lyons**
CORE-19579

## Purpose

_Move CMD popup menu to not cover 'Close'._

JIRA issue: [CORE-19579](https://jira.reactos.org/browse/CORE-19579)

## Proposed changes

_When IS_SYSTEM_MENU was redefined, the old definition worked better in this location, so put its logic back._
_Fix an 'unintended consequence' of https://github.com/reactos/reactos/commit/3ae0ccdcc._

Before:

cmd-taskbar-right-click-close-bad

After:

cmd-menu-pos-fix1

### CODE REVIEW

**Comment by JoachimHenze** [comment]
@Doug-Lyons does https://jira.reactos.org/browse/CORE-19061 remain in "fixed"-state after your PR?

**Comment by Doug-Lyons** [comment]
> @Doug-Lyons does https://jira.reactos.org/browse/CORE-19061 remain in "fixed"-state after your PR?

Yes. See Below:

taskmgr_rt_clk_on_titlebar

**Comment by HBelusca** [comment]
@Cleverking2003 Could you check this PR since you also previously worked on fixing some window system menu handling?

**Comment by HBelusca** [comment]
@Doug-Lyons Regarding:
> Partial revert of redefinition of IS_SYSTEM_MENU.
The guilty revision was https://github.com/reactos/reactos/commit/3ae0ccdcc.

I'm not sure how this fully relates to a partial redefinition or the PR, since the IS_SYSTEM_MENU macro hasn't been touched in the PR.

**Review by JoachimHenze — APPROVED** [review]
Approved. Thanks for retesting the Taskmgr as well. Regarding Hermes comment: Yes, maybe "partial revert of..." could be rephrased/precised a bit.

**Comment by Doug-Lyons** [comment]
> @Doug-Lyons Regarding:
> 
> > Partial revert of redefinition of IS_SYSTEM_MENU.
> > The guilty revision was [3ae0ccdcc](https://github.com/reactos/reactos/commit/3ae0ccdcc).
> 
> I'm not sure how this fully relates to a partial redefinition as quoted by the commit, since the IS_SYSTEM_MENU macro hasn't been touched in the PR.

This uses IS_SYSTEM_MENU and worked fine before its redefinition. Now I have restored the old logic before IS_SYSTEM_MENU was redefined for THIS logic only. I have reworded the description to hopefully clarify this a bit more. Thanks.

**Comment by HBelusca** [comment]
I would like to have @Cleverking2003 have a look at it before giving approval.

**Review by cbialorucki — APPROVED** [review]

**Comment by Doug-Lyons** [comment]
> I would like to have @Cleverking2003 have a look at it before giving approval.

@HBelusca, We can certainly wait a few days before merging to see if @Cleverking2003 might have any comments. Thanks.

**Review by DarkFire01 — APPROVED** [review]

**Comment by Cleverking2003** [comment]
LGTM. The system menu logic is a bit messy and that's probably why I accidentally removed that check.

**Review by HBelusca — APPROVED** [review]

**Comment by Doug-Lyons** [comment]
Thanks for everyone's help with this.

**Pull request merged**
